src/libraries/Microsoft.PowerFx.Core/Public/Engine.cs

```diff
@@ -155,6 +186,12 @@ private CheckResult CheckInternal(ParseResult parse, RecordType parameterType, B
                 result.Expression = CreateEvaluator(result);
             }
 
+            var endHash = symbolTable?.VersionHash;
+            if (startHash != endHash)
```

Comment:
Shouldn't this be in a Contracts.Assert(...)

Reply:
no, it's a runtime error, and I also include a test to verify. 
(Contract.Assert can get optimized out in retail builds)

This is similar to "can't mutate a collection while enumerating it" exceptions.